packages/react/src/provider.tsx

```diff
@@ -72,6 +73,14 @@ export const LogtoProvider = ({ logtoConfig, children }: LogtoProviderProperties
       try {
         await logtoClient.handleCallback(uri);
         dispatch({ type: 'HANDLE_CALLBACK_SUCCESS', payload: { claims: logtoClient.getClaims() } });
+        if (window.location.href.includes('?')) {
+          const { url: baseUrl, query: queryObject } = queryString.parseUrl(window.location.href);
```

Comment:
Option 1: (if you want to use `query-string` as a dependency)

this whole block equals to:

```js
const filterUrl = queryString.exclude(uri, ['code', 'state', 'error', 'error_description']);
history.replaceState(history.state, '', filterUrl);
```

However we could avoid introducing extra `query-string`  dependency.
Option 2: recommended without any add-ons

```js
import { useHistory } from 'react-router-dom';

const history = useHistory();
 const queries = new URLSearchParams(window.location.search);
        
queries.delete('code');
queries.delete('state');

history.push({ search: queries.toString() });

```

Reply:
last thing. instead of hard code the params key `code`, `state`, (`error`, `error_description` BTW), should import the keys from the `client` sdk  `AuthenticationResult` type. 

using `https://github.com/kimamula/ts-transformer-keys` or  simply define a const array but type is guarded by keyof `AuthenticationResult`